program_intent_engagement/apps/api/v1/views.py

```diff
@@ -1 +1,85 @@
-# Create your views here.
+"""
+V1 API Views
+"""
+from itertools import groupby
+
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from program_intent_engagement.apps.api.serializers import ProgramIntentSerializer
+from program_intent_engagement.apps.core.models import ProgramIntent
+
+
+class MostRecentAndCertainIntentsView(APIView):
+    """
+    Retrieve a list of user intents
+    This endpoint returns a list of intents specified by the user.
+      Grouped by programs and sorted by most recent
+      and CERTAIN (if available, otherwise MAYBE) intent
+    Path: /api/[version]/intents/most-recent-and-certain
+    Returns:
+     * 200: OK, list of intents
+     * 404: User not found
+    """
+    authentication_classes = (JwtAuthentication,)
+    serializer_class = ProgramIntentSerializer
+
+    @classmethod
+    def grab_best_intents(cls, list_of_intents):
+      """Given a list of intents returns a dictionary of best intent per program
+
+      Args:
+          list_of_intents (_type_): A list of intents(dictionaries)
```

Comment:
intents are actually ProgramIntent objects not just dictionaries

Reply:
They're converted to a list of dictionaries through .values() which is how i'm able to use groupby is how i thought of it

Reply:
Or I could take change it to take in program intents by user and then do the .orderBy and .values in this method

Reply:
taking in the list of intents is an easier interface than taking in a specially pre-chewed version of the intents. It moves a little work in here to make the interface simpler.